src/mylibopencm3.c

```diff
@@ -0,0 +1,61 @@
+#include "mylibopencm3.h"
+
+void rcc_clock_setup_hsi_3v3(const struct rcc_clock_scale *clock)
+{
+	/* Enable internal high-speed oscillator (HSI). */
+	rcc_osc_on(RCC_HSI);
+	rcc_wait_for_osc_ready(RCC_HSI);
+
+	/* Select HSI as SYSCLK source. */
+	rcc_set_sysclk_source(RCC_CFGR_SW_HSI);
+
+	/* Set the VOS scale mode */
+	rcc_peripheral_enable_clock(&RCC_APB1ENR, RCC_PWR);
+	pwr_set_vos_scale(clock->voltage_scale);
+
+	/*
+	 * Set prescalers for AHB, ADC, ABP1, ABP2.
+	 * Do this before touching the PLL (TODO: why?).
+	 */
+	rcc_set_hpre(clock->hpre);
```

Comment:
This is copied from libopencm3. I will work on this to avoid having to duplicate libopencm3 code in our project: https://github.com/libopencm3/libopencm3/pull/1032

Will probably wait until April 27th though. :wink:

Reply:
ok